### ISSUE TITLE
Store `userId` in client side and verify with `jwt` with every api call

### ISSUE THREAD

**Issue by ricktjwong** [opengovsg/ccrm#50]
There are two design choices in this api routing case. We could either do something like `users/self/cases` as opposed to something like `users/:id/cases`. Choosing the former would invalidate the need to verify the `userId` with every api call, but its design might hinder the more general use of routes, for example in a search functionality where one would want to see some details of other cases that do not belong to them.

tldr:
using the latter api route `users/:id/cases` incurs greater implementation complexity but this is chosen for its better modularity.
